src/clib/lib/enkf/read_summary.cpp

```diff
@@ -1,25 +1,58 @@
+#include <algorithm>
 #include <ert/ecl/ecl_smspec.hpp>
 #include <ert/ecl/ecl_sum.hpp>
 #include <ert/python.hpp>
 #include <fnmatch.h>
+#include <pybind11/pybind11.h>
+#include <stdexcept>
 #include <string>
+#include <time.h>
 #include <tuple>
 #include <vector>
 
+#include <datetime.h> // must be included after pybind11.h
+
 static bool matches(std::vector<std::string> patterns, std::string key) {
-    bool has_key = false;
-    for (auto pattern : patterns) {
-        if (fnmatch(pattern.c_str(), key.c_str(), 0) == 0) {
-            has_key = true;
-            break;
-        }
-    }
-    return has_key;
+    return std::any_of(patterns.cbegin(), patterns.cend(),
+                       [&key](const std::string &pattern) {
+                           return fnmatch(pattern.c_str(), key.c_str(), 0) == 0;
+                       });
 }
+
 ERT_CLIB_SUBMODULE("_read_summary", m) {
+    m.def("read_dates", [](Cwrap<ecl_sum_type> summary) {
+        if (!PyDateTimeAPI)
+            PyDateTime_IMPORT;
+
+        time_t_vector_type *tvec = ecl_sum_alloc_time_vector(summary, true);
+        int size = time_t_vector_size(tvec);
+        std::vector<pybind11::object> result(size);
```

Comment:
Maybe use a `py::list` directly? No need to use an `std::vector` if it ends up as a Python `list` in the end anyway.